flair/training_utils.py

```diff
@@ -15,64 +17,94 @@ class Metric(object):
+    def f_score(self, cls=None):
+        if self.precision(cls) + self.recall(cls) > 0:
+            return round(2 * (self.precision(cls) * self.recall(cls)) / (self.precision(cls) + self.recall(cls)), 4)
         return 0.0
 
-    def accuracy(self):
-        if self._tp + self._tn + self._fp + self._fn > 0:
-            return round((self._tp + self._tn) / (self._tp + self._tn + self._fp + self._fn), 4)
+    def accuracy(self, cls=None):
+        if self._tps[cls] + self._tns[cls] + self._fps[cls] + self._fns[cls] > 0:
+            return round(
+                (self._tps[cls] + self._tns[cls]) / (self._tps[cls] + self._tns[cls] + self._fps[cls] + self._fns[cls]),
+                4)
         return 0.0
 
     def to_tsv(self):
+        # gather all the classes
         return '{}\t{}\t{}\t{}\t{}\t{}\t{}\t{}'.format(
-            self._tp, self._tn, self._fp, self._fn, self.precision(), self.recall(), self.f_score(), self.accuracy())
+            self.get_tp(), self.get_tn(), self.get_fp(), self.get_fn(), self.precision(), self.recall(), self.f_score(), self.accuracy())
 
     def print(self):
         log.info(self)
 
     @staticmethod
     def tsv_header(prefix=None):
         if prefix:
-            return '{0}_TP\t{0}_TN\t{0}_FP\t{0}_FN\t{0}_PRECISION\t{0}_RECALL\t{0}_F-SCORE\t{0}_ACCURACY'.format(prefix)
+            return 'CLS\t{0}_TP\t{0}_TN\t{0}_FP\t{0}_FN\t{0}_PRECISION\t{0}_RECALL\t{0}_F-SCORE\t{0}_ACCURACY'.format(
+                prefix)
 
-        return 'TP\tTN\tFP\tFN\tPRECISION\tRECALL\tF-SCORE\tACCURACY'
+        return 'CLS\TP\tTN\tFP\tFN\tPRECISION\tRECALL\tF-SCORE\tACCURACY'
 
     @staticmethod
     def to_empty_tsv():
         return '_\t_\t_\t_\t_\t_\t_\t_'
 
     def __str__(self):
-        return '{0:<10}\ttp: {1} - fp: {2} - fn: {3} - tn: {4} - precision: {5:.4f} - recall: {6:.4f} - accuracy: {7:.4f} - f1-score: {8:.4f}'.format(
-            self.name, self._tp, self._fp, self._fn, self._tn, self.precision(), self.recall(), self.accuracy(), self.f_score())
+        all_classes = self.get_classes()
+        all_lines = [
+            '{0:<10}\ttp: {1} - fp: {2} - fn: {3} - tn: {4} - precision: {5:.4f} - recall: {6:.4f} - accuracy: {7:.4f} - f1-score: {8:.4f}'.format(
+                MICRO_AVG_METRIC if cls == None else cls,
+                self._tps[cls], self._fps[cls], self._fns[cls], self._tns[cls],
+                self.precision(cls), self.recall(cls), self.accuracy(cls), self.f_score(cls))
+            for cls in all_classes]
+        return '\n'.join(all_lines)
+
+    def get_classes(self) -> List:
+        all_classes = list(set(itertools.chain(*[list(keys) for keys
+                                                 in [self._tps.keys(), self._fps.keys(), self._tns.keys(),
+                                                     self._fns.keys()]])))
+
+        all_classes.sort(key=lambda x: (x is not None, x))
+        return all_classes
```

Comment:
I would love to see methods to calculate the micro and macro average for all classes. Do you think you can add those? Otherwise I'll do it later on.

Reply:
They're added as methods to `Metric`